components/org.wso2.carbon.identity.auth.valve/src/main/java/org/wso2/carbon/identity/auth/valve/util/APIErrorResponseHandler.java

```diff
@@ -85,6 +91,18 @@ public static void handleErrorResponse(AuthenticationContext authenticationConte
         }
     }
 
+    public static void handleAuthenticationFailErrorResponse(AuthenticationContext authenticationContext, Response response,
+                                                       int error, AuthenticationFailException e) throws IOException {
+
+        if (e != null && e.getErrorCode() != null &&
+                USER_ACCOUNT_LOCKED_ERROR_CODE.equals(e.getErrorCode().split(":")[0]) && Boolean.parseBoolean(
+                IdentityUtil.getProperty(IdentityConstants.APIResponse.SET_ACCOUNT_LOCK_AUTH_FAILURE_REASON))) {
```

Comment:
Lets add the SET_ACCOUNT_LOCK_AUTH_FAILURE_REASON property check as the first boolean value since it's the main dependent value. The string operation and other checks need not run if the config is set to false.

Reply:
Resolves via https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/260/commits/1de695ba94b57d597ad96b806b45e31c797669fb